armory/docker/images.py

```diff
@@ -32,3 +36,218 @@
 # way to find out
 def is_old(tag: str):
     raise NotImplementedError
+
+
+def split_name(image_name: str):
+    """
+    Parse image name and return tuple (repo, name, tag)
+        Return the empty string "" for any that do not exist
+    """
+    if not image_name:
+        raise ValueError("image_name cannot be empty")
+
+    tokens = image_name.split(":")
+    if len(tokens) == 1:
+        repo_name = tokens[0]
+        tag = ""
+    elif len(tokens) == 2:
+        repo_name, tag = tokens
+        if "/" in tag:
+            raise ValueError(
+                f"invalid docker image image_name {image_name}. '/' cannot be in tag"
+            )
+    else:
+        raise ValueError(f"invalid docker image image_name {image_name}. Too many ':'")
+
+    if "/" in repo_name:
+        i = repo_name.rindex("/")
+        repo = repo_name[:i]
+        name = repo_name[i + 1 :]
+    else:
+        repo = ""
+        name = repo_name
+    if not name:
+        raise ValueError(f"invalid docker image image_name {image_name}. No name")
+
+    return repo, name, tag
+
+
+def join_name(repo: str, name: str, tag: str):
+    """
+    Inverse of split_image_name. The following should be identity operations:
+        image_name = join_name(*split_name(image_name))
+        repo, name, tag = split_name(join_name(repo, name, tag))
+    """
+    if not name:
+        raise ValueError("name cannot be empty")
+    if repo:
+        repo = repo + "/"
+    if tag:
+        tag = ":" + tag
+    return f"{repo}{name}{tag}"
+
+
+def is_armory(image_name: str):
+    """
+    Return whether image_name refers to an armory docker image
+    """
+    repo, name, _ = split_name(image_name)
```

Comment:
technically `user` and `repository` (I just checked when writing my split_name above.

Reply:
Good! I was looking for the correct naming convention.